### ISSUE TITLE
"Storage is not configured correctly" reported and function works

### ISSUE THREAD

**Issue by tymtam2** [Azure/Azure-Functions#2169]
Runtime version: 4.1.3.17473

All my v4 function apps complain that "Storage is not configured correctly", and they all work. 

![image](https://user-images.githubusercontent.com/1097613/155478164-ae45c291-42a3-4137-b19e-d37ac20d7c02.png)

I have followed https://docs.microsoft.com/en-us/azure/azure-functions/functions-recover-storage-account and confirmed that the storage account is accessible via the shared key.

All functions have `AzureWebJobsStorage` setting set.

**Comment by v-bbalaiagar** [comment]
Hi @tymtam2 , Thank you for your feedback! Here are the steps how you can resolve the azure function app error "storage is not configured" - https://www.connected-pawns.com/2020/06/14/azure-function-apps-storage-is-not-configured/

Please check and let us know if this helps.

**Comment by tymtam2** [comment]
@v-bbalaiagar Thank you very much for you reply. I will add the settings to my resource creation script. 

It's worth pointing out that [App settings reference for Azure Functions](https://docs.microsoft.com/en-us/azure/azure-functions/functions-app-settings#website_contentshare) states: 

> When using an Azure Resource Manager template to create a function app during deployment, don't include WEBSITE_CONTENTSHARE in the template. This slot setting is generated during deployment. To learn more, see [Automate resource deployment for your function app](https://docs.microsoft.com/en-us/azure/azure-functions/functions-infrastructure-as-code?tabs=windows#create-a-function-app).

When you follow the link the template actually contains WEBSITE_CONTENTSHARE but then a note below says: 
> Important
> 
> Do not need to set the [WEBSITE_CONTENTSHARE](https://docs.microsoft.com/en-us/azure/azure-functions/functions-app-settings#website_contentshare) setting in a deployment slot. This setting is generated for you when the app is created in the deployment slot.

I find it confusing. 

Thanks again!

**Comment by tymtam2** [comment]
After auditing my functions it seem that the this problem affects only the functions created using bicep templates.

All functions deployed using Azure CLI have WEBSITE_CONTENTAZUREFILECONNECTIONSTRING and WEBSITE_CONTENTSHARE set without me explicitly setting it. 

Example CLI: 
```
az functionapp create  `
  --subscription "$subscription" `
  --resource-group "$resource_group" `
  --name $functionAppName `
  --storage-account $storageName `
  --consumption-plan-location "$functionLocation" `
  --resource-group "$resource_group" `
  --functions-version 4
```

**Comment by lokijota** [comment]
I have exactly the same problem. Non-premium functions, deployed with ARM.  Only one of 3 functions complains about the error, weirdly

**Comment by cghgreg** [comment]
I received this warning when my app config didn't have the WEBSITE_CONTENTAZUREFILECONNECTIONSTRING and WEBSITE_CONTENTSHARE settings. It also appears that publishing using the AzureFunctionApp Azure Pipelines task does not create these settings automatically. After I added these settings manually the warning went away.

The documentation makes it sounds like these settings are only required for premium consumption, which I took to mean an Elastic Premium SKU, but this appears to not be the case.

Can the documentation be corrected or made more clear if this is true? Or is this a bug in Azure Portal that this warning is shown when it shouldn't be? The app seemed to be functioning fine while the warning was visible.
